### ISSUE TITLE
Add a C API to neuropod

### ISSUE THREAD

**Issue by vkuzmin-uber** [uber/neuropod#407]
## Feature
C API

We don't expect that somebody is using C API directly. This will be used by future CGo API mostly.

Connected to https://github.com/uber/neuropod/issues/294

## Describe the solution you'd like
Design is inspired by the TensorFlow C API and Uber CGO package that includes Bridge C/C++ bridge. Namely:
Internal C Headers
Public C Headers
C/C++ implementation of public and internal headers.

**Comment by VivekPanyam** [comment]
> We don't expect that somebody is using C API directly. This will be used by future CGo API mostly.

I don't think this is accurate. We should build this API in a way where it provides a robust easy-to-use interface to use Neuropod from C (independent of the Go API)

**Comment by vkuzmin-uber** [comment]
> > We don't expect that somebody is using C API directly. This will be used by future CGo API mostly.
> 
> I don't think this is accurate. We should build this API in a way where it provides a robust easy-to-use interface to use Neuropod from C (independent of the Go API)

Agree, just wanted to highlight that CGO is going to be a first user really, not only.
